modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -5011,6 +5020,40 @@ public static long nextDumpTimeout(int step, long timeout) {
         return nextTimeout <= limit ? nextTimeout : limit;
     }
 
+    /**
+     * If partition is clearing or already cleared we need full rebalance even if supplier is exists.
+     * (it still could be used for other demanders)
+     *
+     * @param grp Group.
+     * @param part Partition.
+     */
+    public boolean isClearingPartition(CacheGroupContext grp, int part) {
```

Comment:
Are you sure that this optimization works correctly?
In case of persistence enabled = true this method will always return false, while addClearingPartition works only if persistence enabled = false.